code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -216,34 +216,30 @@
 
 /// Extra handling for adding the action for draggin functionality (for instant building)
 /datum/action/xeno_action/activable/secrete_resin/give_action(mob/living/L)
-	..()
+	. = ..()
 	if(!(CHECK_BITFIELD(SSticker.mode.flags_round_type, MODE_ALLOW_XENO_QUICKBUILD) && SSresinshaping.active))
 		return
+
+	var/mutable_appearance/build_maptext = mutable_appearance(icon = null,icon_state = null, layer = ACTION_LAYER_MAPTEXT)
+	build_maptext.pixel_x = 12
+	build_maptext.pixel_y = -5
+	build_maptext.maptext = MAPTEXT(SSresinshaping.get_building_points(owner))
+	visual_references[VREF_MUTABLE_BUILDING_COUNTER] = build_maptext
+
 	RegisterSignal(owner, COMSIG_MOB_MOUSEDOWN, .proc/start_resin_drag)
 	RegisterSignal(owner, COMSIG_MOB_MOUSEDRAG, .proc/preshutter_resin_drag)
 	RegisterSignal(owner, COMSIG_MOB_MOUSEUP, .proc/stop_resin_drag)
 	RegisterSignal(SSdcs, list(COMSIG_GLOB_OPEN_SHUTTERS_EARLY, COMSIG_GLOB_OPEN_TIMED_SHUTTERS_LATE,COMSIG_GLOB_OPEN_TIMED_SHUTTERS_XENO_HIVEMIND,COMSIG_GLOB_TADPOLE_LAUNCHED,COMSIG_GLOB_DROPPOD_LANDED), .proc/end_resin_drag)
 
 /// Extra handling to remove the stuff needed for dragging
 /datum/action/xeno_action/activable/secrete_resin/remove_action(mob/living/carbon/xenomorph/X)
-	..()
-	if(!CHECK_BITFIELD(SSticker.mode.flags_round_type, MODE_ALLOW_XENO_QUICKBUILD))
+	. = ..()
+	if(!(CHECK_BITFIELD(SSticker.mode.flags_round_type, MODE_ALLOW_XENO_QUICKBUILD)) && SSresinshaping.active)
```

Comment:
this is a bad idea , since due to signals being async and not having a set order , its possible for the SSresinshaping to disable itself first, and then not allow removal of the features due to SSresinshaping not being active.

Reply:
That makes sense actually. 
```suggestion
	if(!(CHECK_BITFIELD(SSticker.mode.flags_round_type, MODE_ALLOW_XENO_QUICKBUILD)))
```